src/components/EditCategoryModal/EditCategoryModal.jsx

```diff
@@ -0,0 +1,180 @@
+}) => {
+  const { data: dataCategories, isFetched: isFetchedCategories } = useQuery(
+    ['categories'],
+    getCategories
+  );
+
+  const categoriesDeleteModal = useDisclosure();
+
+  const {
+    register,
+    handleSubmit,
+    control,
+    reset,
+    isDirty,
+    formState: {
+      errors: { name }
+    }
+  } = useForm({
+    defaultValues: {
+      categoryType: categoryData.categoryType,
+      categoryId: categoryData.id,
+      name: categoryData.name,
+      color: categoryData.color
+    }
+  });
+  const resetForm = () => {
+    reset({
+      categoryType: categoryData.categoryType,
+      categoryId: categoryData.id,
+      name: categoryData.name,
+      color: categoryData.color
+    });
+  };
+  useEffect(() => resetForm(), [!isOpen]);
+
+  const closeAllModals = () => {
+    categoriesDeleteModal.onClose();
+    onClose();
+  };
+  //validation does not work, because the form already has default fields, and when compared in isDirty, it considers that everything has been changed
+  const onCancel = () => {
+    isDirty ? categoriesDeleteModal.onOpen() : onClose();
+    // onClose();
+  };
+
+  return (
+    <>
+      <Modal
+        isOpen={isOpen}
+        onClose={categoriesDeleteModal.onOpen}
+        closeOnOverlayClick={false}
+      >
+        <ModalOverlay />
+        <ModalContent>
+          <ModalHeader>
+            {categoryType === 'Income'
+              ? i18next.t('modal.editCategory.header.incomeCategory')
+              : i18next.t('modal.editCategory.header.expensesCategory')}
+          </ModalHeader>
+          <ModalCloseButton />
+          <ModalBody>
+            <FormControl isRequired isInvalid={name}>
+              <Flex>
+                <FormLabel htmlFor="name" w="40%">
+                  {i18next.t('modal.addCategory.formName')}
+                </FormLabel>
+                <Flex flexDirection="column">
+                  <Input
+                    {...register('name', {
+                      required: i18next.t(
+                        'modal.addCategory.validationErrorMessage.name'
+                      ),
+                      maxLength: {
+                        value: 100,
+                        message: i18next.t(
+                          'modal.addCategory.validationErrorMessage.nameIsLong'
+                        )
+                      },
+                      pattern: {
+                        value: /^[a-zA-Z0-9$@$!%*?&#^-_. +]+$/,
+                        message: i18next.t(
+                          'modal.addCategory.validationErrorMessage.pattern'
+                        )
+                      },
+                      validate: (name) =>
+                        (isFetchedCategories &&
+                          !dataCategories.data
+                            .filter(
+                              (data) => data.categoryType === categoryType
+                            )
+                            //.filter((data) => data.id !== categoryData.id)
```

Comment:
remove?

Reply:
here I asked the backend to correct their validation and we could change only the color or only the name, I'm waiting for them to do this.